packages/recommend-core/src/getBatchRecommendations.ts

```diff
@@ -55,10 +56,18 @@ export async function getBatchRecommendations<TObject>({
     const splitResult = response?.results?.slice(prevChunks, allChunks);
     prevChunks += keyPair.value;
 
-    let recommendations: Array<ProductRecord<ProductRecord<TObject>>>;
-
-    if (model === 'trending-facets' || model === 'trending-items') {
-      recommendations = mapByScoreToRecommendations<TrendingFacet<TObject>>({
+    let recommendations: Array<ProductRecord<ProductRecord<TObject>>> = [];
+    let trendingFacets: Array<TrendingFacet<TFacet>> = [];
+    if (model === 'trending-facets') {
+      trendingFacets = mapByScoreToRecommendations<TrendingFacet<TFacet>>({
+        maxRecommendations,
+        hits: splitResult
+          .map((res) => res.hits)
+          .flat()
+          .map((hit) => (hit as unknown) as TrendingFacet<TFacet>),
```

Comment:
not super happy about this line

Reply:
Do we even need to run this function on facets?

Anyway we shouldn't need to loop over the items to cast `hits`.

Reply:
Good question 👀 indeed this is not useful since the goal of this function is to handle when we pass multiple queries to the component, however in our case we can't pass multiple facetNames ! 


Really good catch !